src/java.base/share/classes/java/lang/invoke/InnerClassLambdaMetafactory.java

```diff
@@ -209,7 +205,7 @@ private static String lambdaClassName(Class<?> targetClass) {
             // use the original class name
             name = name.replace('/', '_');
         }
-        return name.replace('.', '/') + "$$Lambda$" + counter.incrementAndGet();
+        return name.replace('.', '/') + "$$Lambda$";
```

Comment:
Using `String::concat` uses the same code path as indy‑based String concatenation[^1].
```suggestion
        return name.replace('.', '/').concat("$$Lambda");
```

[^1]: The JDK uses `StringBuilder`‑based String concatenation:
https://github.com/openjdk/jdk/blob/a39cf2e3b242298fbf5fafdb8aa9b5d4562061ef/make/modules/java.base/Java.gmk#L28

Reply:
The original code used `+` so I assume it's not a problem here... but if I'm wrong, wouldn't it be better to migrate from `+` to `concat` in a separate change?